test/suites/dev/test-xcm/test-xcm-erc20-v3.ts

```diff
@@ -0,0 +1,135 @@
+import { beforeEach, describeSuite, expect } from "@moonwall/cli";
+import { ALITH_ADDRESS, CHARLETH_ADDRESS, alith } from "@moonwall/util";
+import { ApiPromise } from "@polkadot/api";
+import { parseEther } from "ethers";
+import { expectEVMResult } from "../../../helpers/eth-transactions.js";
+import {
+  XcmFragment,
+  XcmFragmentConfig,
+  injectHrmpMessage,
+  sovereignAccountOfSibling,
+} from "../../../helpers/xcm.js";
+
+export const ERC20_TOTAL_SUPPLY = 1_000_000_000n;
+
+describeSuite({
+  id: "D2705",
+  title: "Mock XCM V3 - Receive erc20 via XCM",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let erc20ContractAddress: string;
+    let polkadotJs: ApiPromise;
+
+    beforeEach(async function () {
+      polkadotJs = context.polkadotJs();
+
+      const { contractAddress, status } = await context.deployContract!("ERC20WithInitialSupply", {
+        args: ["ERC20", "20S", ALITH_ADDRESS, ERC20_TOTAL_SUPPLY],
+      });
+      erc20ContractAddress = contractAddress;
+      expect(status).eq("success");
+    });
+
+    it({
+      id: "T01",
+      title: "Should be able to transfer ERC20 token through incoming XCM message",
+      test: async function () {
+        const paraId = 888;
+        const paraSovereign = sovereignAccountOfSibling(context, paraId);
+        const amountTransferred = 1_000_000n;
+
+        // Get pallet indices
+        const metadata = await polkadotJs.rpc.state.getMetadata();
+        const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
```

Comment:
Same comment as before, interact with object directly without casting in order to preserve type safety